File: test-corpus/legitimate/python/001-api-client.py
# Legitimate API client - should PASS
# Expected: PASS or WARN (acceptable)
# Description: Safe API usage with environment variables

import requests
import os

API_KEY = os.environ.get('API_KEY', '')
response = requests.get('https://api.example.com/data',
                       headers={'Authorization': f'Bearer {API_KEY}'})
print(response.json())

File: test-corpus/legitimate/typescript/001-api-client.ts
// Legitimate API client - should PASS
// Expected: PASS or WARN (acceptable)
// Description: Safe API usage with environment variables

export async function fetchUserData(userId: string) {
  const apiKey = process.env.GITHUB_TOKEN || '';
  const response = await fetch(`https://api.github.com/users/${userId}`, {
    headers: {
      'Authorization': `Bearer ${apiKey}`,
      'User-Agent': 'my-app'
    }
  });
  return response.json();
}
